### DISCUSSION TITLE
Custom visualization documentation

### DISCUSSION THREAD

**Discussion by madeloop** [perspective-dev/perspective#2188] [Q&A]
Is there up-to-date documentation/blog post/video about how to go about creating a custom visualization plugin?

**Comment by texodus** [comment]
[https://github.com/finos/perspective/blob/master/rust/perspective-viewer/README.md#interface-iperspectiveviewerplugin](https://github.com/finos/perspective/blob/master/rust/perspective-viewer/README.md#interface-iperspectiveviewerplugin)
[https://github.com/finos/perspective/blob/master/rust/perspective-viewer/src/ts/plugin.ts](https://github.com/finos/perspective/blob/master/rust/perspective-viewer/src/ts/plugin.ts)

**Comment by timkpaine** [comment]
I have created a sample plugin based on @texodus’s links above:
https://github.com/perspective-community/perspective-viewer-example-plugin

Note that despite my poor logo design skills, this example is **not** maintained or supported by the perspective team, just something I created as a one-off to work on a few proof-of-concept plugins on the side. It may break or become incompatible as perspective development continues, and substantial plugins are encouraged to be PR’d direct to perspective (this is how highcharts, d3fc, openlayers, etc, were contributed).
